File: frontend/lib/models/plan/plan_update_request.dart
class PlanUpdateRequest {
  final String title;
  final String description;
  final DateTime startDate;
  final DateTime endDate;

  PlanUpdateRequest({
    required this.title,
    required this.description,
    required this.startDate,
    required this.endDate,
  });

  Map<String, dynamic> toJson() {
    return {
      'title': title,
      'description': description,
      'startDate': startDate.toIso8601String(),
      'endDate': endDate.toIso8601String(),
    };
  }
}

File: backend/src/main/java/com/patriot/fourlipsclover/plan/dto/request/PlanUpdateRequest.java
package com.patriot.fourlipsclover.plan.dto.request;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class PlanUpdateRequest {

    private String title;
    private String description;
    private LocalDate startDate;
    private LocalDate endDate;

}
